swyft/store/store.py

```diff
@@ -191,11 +191,14 @@ def log_lambda(self, z: np.ndarray) -> np.ndarray:
             d = np.where(r > d, r, d)
         return d
 
-    def sample(self, N, pdf):
+    def sample(self, N, pdf, exactly_n: bool = False):
         self._update()
 
         # Generate new points
-        z_prop = pdf.sample(N=np.random.poisson(N))
+        if exactly_n:
+            z_prop = pdf.sample(N=N)
```

Comment:
This only guarantees fixed sample numbers if the store is empty.

Reply:
addressed in the other location. Let's talk there.